osu.Game/Database/ModelManager.cs

```diff
@@ -81,16 +81,19 @@ public void ReplaceFile(RealmNamedFileUsage file, Stream contents, Realm realm)
         }
 
         /// <summary>
-        /// Add a file from within an ongoing realm transaction. If the file already exists, it is overwritten.
+        /// Add a file from within an ongoing realm transaction. If the file already exists, it is overwritten so long as <paramref name="overwrite"/> is true.
         /// </summary>
-        public void AddFile(TModel item, Stream contents, string filename, Realm realm)
```

Comment:
As for why I touched this

*From [`b7883f1` (#30226)](https://github.com/ppy/osu/pull/30226/commits/b7883f18be12d304fddb57269c5bcb123e0763b2)*

> The GetFile method in AddFile has a huge overhead, given we're doing
> this in a loop.
> 
> Since we clear the files in the skin, we already know there won't be any
> existing files, so we can skip all of that logic

Reply:
This makes no sense and offers zero safeties.

If the thing has too high overhead, we fix later. Not as part of a huge change. Remove this commit please.